internal/blockprocessor/processor.go

```diff
@@ -133,9 +134,11 @@ func (b *BlockProcessor) Start() {
 				continue
 			}
 
-			if err = b.listeners.invoke(block); err != nil {
-				panic(err)
-			}
+			go func(curBlock *types.Block) {
+				if err = b.listeners.invoke(curBlock); err != nil {
+					panic(err)
+				}
+			}(block)
```

Comment:
You cant start the goroutine here, those are general listeners that not all will want to behave async like that. I think it is better start it in the PendingTx.DoneWithReceipt(...) method.

Reply:
I moved the go routine invocation to `transaction_processor.go`.
FYI, currently, there is only one listener implementation in the code, and this is the `transactionProcessor` which invokes the promises' `DoneWithReceipt()`.